osu.Game/Modes/Mod.cs

```diff
@@ -35,234 +35,129 @@ public abstract class Mod
     }
 
     public abstract class ModHalfTime : Mod
     {
-        public override Mods Name => Mods.HalfTime;
+        public override string Name => "Half Time";
         public override FontAwesome Icon => FontAwesome.fa_osu_mod_halftime;
-        public override string Description => @"Less zoom";
+        public override string Description => "Less zoom";
         public override bool Ranked => true;
-        public override Mods[] DisablesMods => new Mods[] { Mods.DoubleTime, Mods.Nightcore };
+        public override Type[] IncompatibleMods => new[] { typeof(ModDoubleTime), typeof(ModNightcore) };
     }
 
     public abstract class ModNightcore : ModDoubleTime
     {
-        public override Mods Name => Mods.Nightcore;
+        public override string Name => "Nightcore";
         public override FontAwesome Icon => FontAwesome.fa_osu_mod_nightcore;
-        public override string Description => @"uguuuuuuuu";
+        public override string Description => "uguuuuuuuu";
     }
 
     public abstract class ModFlashlight : Mod
     {
-        public override Mods Name => Mods.Flashlight;
+        public override string Name => "Flashlight";
         public override FontAwesome Icon => FontAwesome.fa_osu_mod_flashlight;
-        public override string Description => @"Restricted view area.";
+        public override string Description => "Restricted view area.";
         public override bool Ranked => true;
     }
 
     public class ModAutoplay : Mod
     {
-        public override Mods Name => Mods.Autoplay;
+        public override string Name => "Autoplay";
         public override FontAwesome Icon => FontAwesome.fa_osu_mod_auto;
-        public override string Description => @"Watch a perfect automated play through the song";
+        public override string Description => "Watch a perfect automated play through the song";
         public override double ScoreMultiplier => 0;
-        public override bool Ranked => false;
-        public override Mods[] DisablesMods => new Mods[] { Mods.Relax, Mods.Autopilot, Mods.SpunOut, Mods.SuddenDeath, Mods.Perfect };
+        public override Type[] IncompatibleMods => new[] { typeof(ModRelax), typeof(ModSuddenDeath), typeof(ModPerfect) };
     }
 
     public abstract class ModPerfect : ModSuddenDeath
     {
-        public override Mods Name => Mods.Perfect;
-        public override FontAwesome Icon => FontAwesome.fa_close;
-        public override string Description => @"SS or quit.";
+        public override string Name => "Perfect";
+        public override string Description => "SS or quit.";
     }
 
     public class ModCinema : ModAutoplay
     {
-        public override Mods Name => Mods.Cinema;
+        public override string Name => "Cinema";
         public override FontAwesome Icon => FontAwesome.fa_osu_mod_cinema;
     }
 
-    [Flags]
-    public enum Mods

```

Comment:
We still need to be able to combine mods into a set of flags for serialization / database. Can you add a helper function that does that by iterating over a list of mods?